### PR TITLE
Add a location page that uses Google Maps API

### PR DESCRIPTION
**Pull request by zianc**
This PR creates a new location page in the portfolio that makes use of the
Google Maps API to display the different places I have lived. It completes the
first half of the Google STEP week 4 walkthroughs. It also corrects a typo 
from PR #3. Other features include HTML elements that fade in upon scroll 
between maps.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -38,7 +38,7 @@
      */
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {        
-        int limit = Integer.parseInt(request.getParameter(Constant.COMMENTS_LIMIT));
+        int limit = Integer.parseInt(request.getParameter(Constants.COMMENTS_LIMIT));
```
Comment (razvanm) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:41]: This doesn't look related to this PR and it looks more like a bug in #3. The right thing would be to fix this in a separate (small) PR.

Comment (zianc) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:41]: Ah, I see. Is this something to keep in mind for the future or should I try and move it to its own separate PR now?

Comment (razvanm) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:41]: You can keep this change this PR but please indicate in the PR description the change. :-)

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -54,3 +54,35 @@ function addCommentsToDOM(limit) {
...
+ */
+function createMap(map_name, latitude, longitude) {
+    const map = new google.maps.Map(
+        document.getElementById(map_name),
+        {center: {lat: latitude, lng: longitude}, zoom: 12}
+    );
+}
+
+/*
+ * Creates set of footprints from one div to the next.
```
Comment (razvanm) [inline portfolio/src/main/webapp/script.js:69]: This comment doesn't help the reader understand what's the output you are aiming for. It would also be good to explain the role of the argument. Also, the `elem_name` seems to be an ID.

Also: shouldn't the the step face in the opposite direction?

Comment (zianc) [reply portfolio/src/main/webapp/script.js]: I figured the steps would go from top to bottom to fit with the scroll :)

File: portfolio/src/main/webapp/style.css [hunk 3]
```diff
@@ -65,6 +65,157 @@ input {
...
+    margin-top: 10rem;
+    align-items: center;
+    padding: 2rem;
+}
+
+#comment-form {
+    display: flex;
+    align-items: center;
+    flex-direction: column;
+    width: 100%
```
Comment (yinggao) [inline portfolio/src/main/webapp/style.css:160]: Missing ';' in the end.

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -66,23 +66,29 @@ function createMap(map_name, latitude, longitude) {
 }
 
 /*
- * Creates set of footprints from one div to the next.
+ * Append set of footprints to the DOM, with direction pointing top to bottom,
```
Comment (razvanm) [inline portfolio/src/main/webapp/script.js:69]:
```suggestion
 * Append a set of footprints to the DOM, with direction pointing top to bottom,
```
